src/clr-angular/data/datagrid/datagrid-pagination.ts

```diff
@@ -3,58 +3,45 @@
-        <div class="pagination-description">
-            <ng-content></ng-content>
-        </div>
-        <ul class="pagination-list" *ngIf="page.last > 1">
-            <li *ngIf="page.current > 1">
-                <button
-                        class="pagination-previous"
-                        (click)="page.previous()"
-                        type="button"></button>
-            </li>
-            <li *ngIf="page.current > 2">
-                <button (click)="page.current = 1" type="button">1</button>
-            </li>
-            <li *ngIf="page.current > 3">...</li>
-            <li *ngFor="let pageNum of middlePages" [class.pagination-current]="pageNum === page.current">
-                <button
-                        *ngIf="pageNum !== page.current; else noButton"
-                        (click)="page.current = pageNum"
-                        type="button">{{pageNum}}
-                </button>
-                <ng-template #noButton>{{pageNum}}</ng-template>
-            </li>
-            <li *ngIf="page.current < page.last - 2">...</li>
-            <li *ngIf="page.current < page.last - 1">
-                <button
-                        (click)="page.current = page.last"
-                        type="button">{{page.last}}
-                </button>
-            </li>
-            <li *ngIf="page.current < page.last">
-                <button
-                        class="pagination-next"
-                        (click)="page.next()"
-                        type="button"></button>
-            </li>
-        </ul>
+    <div class="pagination-size" *ngIf="pageSizeComponent">
+      <ng-content select="clr-dg-page-size"></ng-content>
+    </div>
+    <div class="pagination-description">
+      <ng-content></ng-content>
+    </div>
+    <div class="pagination-list" *ngIf="page.last > 1">
+      <button class="pagination-first" [disabled]="page.current === 1" (click)="page.current = 1">
+        <clr-icon shape="angle-double left"></clr-icon>
+      </button>
+      <button class="pagination-previous" [disabled]="page.current === 1" (click)="page.current = page.current - 1">
+        <clr-icon shape="angle left"></clr-icon>
+      </button>
+      <input type="text" class="pagination-current" [size]="page.last.toString().length" [(ngModel)]="page.current" />&nbsp;/&nbsp;<span>{{page.last}}</span>
+      <button class="pagination-next" [disabled]="page.current === page.last" (click)="page.current = page.current + 1">
+        <clr-icon shape="angle right"></clr-icon>
+      </button>
+      <button class="pagination-last" [disabled]="page.current === page.last" (click)="page.current = page.last">
+        <clr-icon shape="angle-double right"></clr-icon>
+      </button>
+    </div>
     `,
   host: { '[class.pagination]': 'true' },
 })
 export class ClrDatagridPagination implements OnDestroy, OnInit {
-  constructor(public page: Page) {}
+  @ContentChild(ClrDatagridPageSize) pageSizeComponent: ClrDatagridPageSize;
```

Comment:
[Our naming conventions](https://github.com/vmware/clarity/blob/master/CODING_GUIDELINES.md#properties-and-methods) say this is `_ pageSizeComponent`. Public because the way Angular works requires it, but prefixed to denote people shouldn't be messing with this.